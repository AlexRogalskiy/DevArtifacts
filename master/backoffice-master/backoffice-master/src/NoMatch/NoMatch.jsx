import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

import {
  Typography,
  withStyles,
} from '@material-ui/core'

const styles = theme => ({
  root: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    maxWidth: '60rem',
    paddingLeft: theme.spacing.unit * 3,
    paddingRight: theme.spacing.unit * 3,
    marginLeft: 'auto',
    marginRight: 'auto',
    height: 'calc(100vh - 104px)',
  },
})

const NoMatch = ({ title, description, classes }) => (
  <div className={classes.root}>
    <div>
      <Typography variant="display3">
        {title}
      </Typography>
      <Typography variant="display1" paragraph>
        {description}
      </Typography>
    </div>
  </div>
)

NoMatch.propTypes = {
  title: PropTypes.string,
  description: PropTypes.node,
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
}

NoMatch.defaultProps = {
  title: '404! Sorry, not found.',
  description: (
    <Fragment>
      This URL does not exist, sorry. Please start over from
      the <Link href="/" to="/">Dashboard</Link>.
    </Fragment>
  ),
}

export default withStyles(styles)(NoMatch)
