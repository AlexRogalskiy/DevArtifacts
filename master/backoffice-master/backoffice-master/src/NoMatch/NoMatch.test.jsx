import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import { mount } from 'enzyme'

import NoMatch from './'

it('renders correctly', () => {
  const tree = mount((
    <Router>
      <NoMatch
        title="Title"
      />
    </Router>
  ))

  expect(tree).toMatchSnapshot()
})

it('renders correctly with description', () => {
  const tree = mount(<NoMatch
    title="Title"
    description={(
      <p>Desc</p>
    )}
  />)

  expect(tree).toMatchSnapshot()
})
