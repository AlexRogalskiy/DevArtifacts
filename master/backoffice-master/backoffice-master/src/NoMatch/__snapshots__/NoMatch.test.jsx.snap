// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="NoMatch-root-1"
>
  <div>
    <h1
      className="MuiTypography-root-2 MuiTypography-display3-4"
    >
      Title
    </h1>
    <p
      className="MuiTypography-root-2 MuiTypography-display1-6 MuiTypography-paragraph-20"
    >
      This URL does not exist, sorry. Please start over from the 
      <a
        href="/"
        onClick={[Function]}
      >
        Dashboard
      </a>
      .
    </p>
  </div>
</div>
`;

exports[`renders correctly with description 1`] = `
<div
  className="NoMatch-root-1"
>
  <div>
    <h1
      className="MuiTypography-root-2 MuiTypography-display3-4"
    >
      Title
    </h1>
    <p
      className="MuiTypography-root-2 MuiTypography-display1-6 MuiTypography-paragraph-20"
    >
      <p>
        Desc
      </p>
    </p>
  </div>
</div>
`;
